builtin/providers/consul/config.go

```diff
@@ -26,9 +30,21 @@ func (c *Config) Client() (*consulapi.Client, error) {
 	if c.Scheme != "" {
 		config.Scheme = c.Scheme
 	}
+
+	tlsConfig := &consulapi.TLSConfig{}
+	tlsConfig.CAFile = c.CAFile
+	tlsConfig.CertFile = c.CertFile
+	tlsConfig.KeyFile = c.KeyFile
+	cc, err := consulapi.SetupTLSConfig(tlsConfig)
+	if err != nil {
+		return nil, err
+	}
+	config.HttpClient.Transport.(*http.Transport).TLSClientConfig = cc
```

Comment:
Here's my thinking:

   1. By [#L37](https://github.com/hashicorp/terraform/pull/7015#discussion-diff-74521998R37), we have a zero-valued object of `consul/api.TLSConfig`, the equivalent of `&consulapi.TLSConfig{}`
   1. Passing the above into [`consul/api.SetupTLSConfig`](https://github.com/hashicorp/consul/blob/c2cbbdec3e236d325363506cc3181b58613483fb/api/api.go#L245) produces a zero-valued `tls.Config`, the equivalent of `&tls.Config{}`
   1.  Assigning the above to `net/http.Transport.TLSClientConfig` does indeed change its default value (`<nil>`), however, according to the [go docs](https://github.com/golang/go/blob/master/src/net/http/transport.go#L113):

> 	// TLSClientConfig specifies the TLS configuration to use with
	// tls.Client. If nil, the default configuration is used.

So, in the end, what we get when we assign an empty `tls.Config` to `net/http.Transport.TLSClientConfig` is the same configuration that Go [chooses at runtime](https://github.com/golang/go/blob/7a622740655bb5fcbd160eb96887032314842e6e/src/net/http/transport.go#L1014) if we didn't do the assignment.